### PR TITLE
EVG-14472: cross-compile to macos arm64

### PR DESCRIPTION
**Pull request by Kimchelly**
JIRA: https://jira.mongodb.org/browse/EVG-14472

* Cross-compile for macos arm64.
* Fix some unrelated backwards-incompatible changes in grip to the JIRA sender.
* `vendor-clean` to remove some riscv64 architecture-specific go files because riscv64 isn't a recognized architecture until go1.16. Since we can't compile with go1.16 yet (not all the build images have it), we have to hack around this by removing the `_riscv64.go` file for now. This should go away when we upgrade to go1.16, after which point we'll presumably have go1.16 available on all the build images.

### CODE REVIEW

File: makefile [hunk 1]
```diff
@@ -171,8 +172,10 @@ vendor-clean:
 	rm -rf vendor/github.com/mongodb/grip/vendor/github.com/google/uuid/
 	rm -rf vendor/github.com/mongodb/grip/vendor/github.com/pkg/
 	rm -rf vendor/github.com/mongodb/grip/vendor/github.com/stretchr/testify/
-	rm -rf vendor/github.com/mongodb/amboy/vendor/github.com/google/uuid
 	rm -rf vendor/github.com/mongodb/grip/vendor/github.com/google/uuid
+	# riscv64 isn't recognized as a valid GOARCH until go1.16, so we can likely stop removing this file once CI tests
```
Comment (julianedwards) [inline makefile:176]: Can you link a ticket?

Comment (julianedwards) [inline makefile:176]: Can you link a ticket? And maybe make it a TODO?

**Review by julianedwards — APPROVED** [review]
LGTM modulo nit

File: makefile (hunk 1 above)
Comment (Kimchelly) [reply makefile]: [EVG-14482](https://jira.mongodb.org/browse/EVG-14482)

**Pull request merged**